src/angular-app/bellows/apps/userprofile/user-profile-app.component.ts

```diff
@@ -203,6 +217,36 @@ export class UserProfileAppController implements angular.IController {
     });
   }
 
+  queryProjectsForUser() {
+    this.finishedLoadingOwnedProjects = false;
+    this.projects = [];
+    this.ownedProjects = [];
+    this.projectService.list().then((projects: Project[]) => {
+      this.projects = projects || [];
+      this.ownedProjects = projects.filter(proj => proj.ownerId === this.user.id);
```

Comment:
@laineyhm The race condition is that `this.loadUser()` might not be done at this point, in which case `this.user.id` is undefined. You can fix it with something like this:



```
    this.loadUser() // load the user data right away
    .then(() => this.queryProjectsForUser());


    private loadUser(): angular.IPromise<unknown> {
        return this.userService.readProfile(result => {
```

Reply:
Great! Thank you. I made this change and tested it. Looks good.